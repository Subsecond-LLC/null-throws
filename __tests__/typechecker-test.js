'use strict';

const path = require('path');
const child_process = require('child_process');

test('nullthrows with flow', () => {
  const ret = child_process.spawnSync(
    'node_modules/.bin/flow',
    ['check', path.join(__dirname, '__fixtures__/flow-package')],
    {
      cwd: path.join(__dirname, '..'),
      encoding: 'utf8',
    }
  );
  expect(ret.stderr).toMatchSnapshot();
  expect(ret.stdout).toMatchSnapshot();
  expect(ret.error).toBeUndefined();
});
