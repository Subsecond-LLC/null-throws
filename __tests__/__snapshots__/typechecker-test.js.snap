// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nullthrows with flow 1`] = `""`;

exports[`nullthrows with flow 2`] = `
"__tests__/__fixtures__/flow-package/index.js:10
 10: var bad1: Object = obj;
                        ^^^ null. This type is incompatible with
 10: var bad1: Object = obj;
               ^^^^^^ object type

__tests__/__fixtures__/flow-package/index.js:10
 10: var bad1: Object = obj;
                        ^^^ undefined. This type is incompatible with
 10: var bad1: Object = obj;
               ^^^^^^ object type


Found 2 errors
"
`;
